crosschain/simple/src/main/java/tech/pegasys/samples/crosschain/simple/transaction/CrosschainTransactionNoParams.java

```diff
@@ -1,4 +1,331 @@
+
+        checkExpectedValues(false);
+
+        Scanner myInput = new Scanner( System.in );
+        Boolean c2Val = this.contract2.get().send();
+
+        LOG.info("  Executing call simulator to determine parameter values and expected results");
+        CallSimulator sim = new CallSimulator();
+        sim.c2clear();
+        sim.c1Crosschain_setter();
+
+        LOG.info("  Constructing Nested Crosschain Transaction");
+        // Originating sidechain is sidechain 1.
+        CrosschainContextGenerator contextGenerator = new CrosschainContextGenerator(SC1_SIDECHAIN_ID);
+
+        // Call to contract 2
+        // Contract 2 is called by contract 1 on sidechain 1.
+        CrosschainContext subordinateContext = contextGenerator.createCrosschainContext(SC1_SIDECHAIN_ID, this.contract1Address);
+        byte[] subordinateTransactionC2 = this.contract2.set_AsSignedCrosschainSubordinateTransaction(subordinateContext);
+
+        // Call to contract 1
+        byte[][] subordinateTransactionsAndViewsForC1;
+        subordinateTransactionsAndViewsForC1 = new byte[][] {subordinateTransactionC2};
+
+        LOG.info("  Executing Crosschain Transaction");
+        // The originating transaction starts in Contract 1.
+        subordinateContext = contextGenerator.createCrosschainContext(subordinateTransactionsAndViewsForC1);
+        transactionReceipt = this.contract1.crosschain_setter_AsCrosschainTransaction(subordinateContext).send();
+        LOG.info("  Transaction Receipt: {}", transactionReceipt.toString());
+        assertTrue(transactionReceipt.isStatusOK());
+
+        boolean stillLocked;
+        final int tooLong = 10;
+        int longTimeCount = 0;
+        StringBuffer graphicalCount = new StringBuffer();
+        do {
+            longTimeCount++;
+            if (longTimeCount > tooLong) {
+                LOG.error("Contract {} did not unlock", this.contract1Address);
+            }
+            Thread.sleep(500);
+            CrosschainIsLocked isLockedObj = this.web3jSc1.crosschainIsLocked(this.contract1Address, DefaultBlockParameter.valueOf("latest")).send();
+            stillLocked = isLockedObj.isLocked();
+            if (stillLocked) {
+                graphicalCount.append(".");
+                LOG.info("   Waiting for the contract to unlock{}", graphicalCount.toString());
+            }
+        } while (stillLocked);
+
+        checkExpectedValues(sim.c2Get());
+
+    }
+
+
+    private void checkExpectedValues(Boolean v1) throws Exception {
+        LOG.info(" Check values have been set as expected");
+        Boolean result;
+
+        result = this.contract2.get().send();
+        LOG.info("  Contract2.val = {}, expecting {}", result.booleanValue(), v1);
+        assertTrue(result.equals(v1));
+    }
+
+
+    private static void assertTrue(boolean val) {
+        if (!val) {
+            throw new Error("Unexpected Result");
+        }
+    }
+
+
+
+
+    private boolean propertiesFileExists() {
```

Comment:
There is a utility to help out with the properties files now. Have a look at:
https://github.com/PegaSysEng/sidechains-samples/blob/master/common/src/main/java/tech/pegasys/samples/sidechains/common/utils/AbstractPropertiesFile.java

Reply:
Done.